tsinfer/inference.py

```diff
@@ -904,6 +907,173 @@ def augment_ancestors(
+    sample_mask=None,
+    site_mask=None,
+    recombination_rate=None,
+    mismatch_ratio=None,
+    path_compression=True,
+    indexes=None,
+    post_process=None,
+    force_sample_times=False,
+    # Deliberately undocumented parameters below
+    recombination=None,  # See :class:`Matcher`
+    mismatch=None,  # See :class:`Matcher`
+    precision=None,
+    extended_checks=False,
+    engine=constants.C_ENGINE,
+    record_provenance=True,
+    map_additional_sites=None,
+):
+    if max_num_partitions is None:
+        max_num_partitions = 1000
+
+    # Convert working_dir to pathlib.Path
+    work_dir = pathlib.Path(work_dir)
+
+    # Create work dir
+    work_dir.mkdir(parents=True, exist_ok=True)
+
+    metadata = {
+        "sample_data_path": str(sample_data_path),
+        "ancestral_allele": ancestral_allele,
+        "sample_mask": sample_mask,
+        "site_mask": site_mask,
+        "ancestor_ts_path": str(ancestor_ts_path),
+        "recombination_rate": recombination_rate,
+        "mismatch_ratio": mismatch_ratio,
+        "path_compression": path_compression,
+        "indexes": indexes,
+        "recombination": recombination,
+        "mismatch": mismatch,
+        "precision": precision,
+        "engine": engine,
+        "extended_checks": extended_checks,
+        "post_process": post_process,
+        "force_sample_times": force_sample_times,
+        "map_additional_sites": map_additional_sites,
+        "record_provenance": record_provenance,
+    }
+    variant_data, ancestor_ts, matcher = load_variant_data_and_ancestors_ts(metadata)
+    sample_indexes = check_sample_indexes(variant_data, indexes).tolist()
+    sample_times = np.zeros(
+        len(sample_indexes), dtype=variant_data.individuals_time.dtype
+    )
+    if force_sample_times:
+        individuals = variant_data.samples_individual[:][sample_indexes]
+        # By construction all samples in an sd file have an individual: but check anyway
+        assert np.all(individuals >= 0)
+        sample_times = variant_data.individuals_time[:][individuals]
+
+        # Here we might want to re-order sample_indexes and sample_times
+        # so that any historical ones come first, any we bomb out early if they conflict
+        # but that would mean re-ordering the sample nodes in the final ts, and
+        # we sometimes assume they are in the same order as in the file
+    sample_times = sample_times.tolist()
+    metadata["sample_indexes"] = sample_indexes
+    metadata["sample_times"] = sample_times
+    num_samples_per_partition = min_work_per_job // variant_data.num_sites
+    if num_samples_per_partition == 0:
+        num_samples_per_partition = 1
+    metadata["num_samples_per_partition"] = num_samples_per_partition
+    metadata["num_partitions"] = math.ceil(
+        len(sample_indexes) / num_samples_per_partition
+    )
+    metadata_path = work_dir / "metadata.json"
+    metadata_path.write_text(json.dumps(metadata))
```

Comment:
handy to indent this too for readability.

Reply:
Fixed in https://github.com/tskit-dev/tsinfer/pull/954/commits/6a833bbd5b2f3cd61eea7fd0d43864f3aaf32a99